### PR TITLE
zkEVM: skeleton menu plus Overview section

### PR DESCRIPTION
**Pull request by kmurphypolygon**
**CURRENT:**

This PR contains the overview section of the architectural material using @omnifient 's diagram as a starting point.

The template menu remains, and is commented out for now with warnings on the empty pages if anyone manages to see one.

**OLD:**

This PR is a draft template for the High Level Architectural documentation we are going to add to the zkEVM docs section, with a view to moving to CDK in the future, and maintaining placeholder items for content not yet ready for publishing.

Use this to build your doc menu items that relate to this epic: https://polygon.atlassian.net/jira/software/projects/DOCS/boards/79/backlog?selectedIssue=DOCS-222

### CODE REVIEW

**Comment by kmurphypolygon** [comment]
I'm gonna put this in dev for you to have a look at @EmpieichO

File: docs/zkEVM/architecture/high-level/overview.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+<!-- https://excalidraw.com/#json=JKZp9QEihifF_B7Z41Dfv,FVNhqQKi9PA1jM0kzUoCsQ" -->
+
+## Intro
+
+This section of the docs looks more closely at the full topology of Polygon's zero-knowledge system architecture.
```
Comment (hsutaiyu) [inline docs/zkEVM/architecture/high-level/overview.md]:
```suggestion
This section provides a detailed analysis of Polygon's zero-knowledge system architecture, exploring its full topology.
```

File: docs/zkEVM/architecture/high-level/overview.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
+<!-- https://excalidraw.com/#json=JKZp9QEihifF_B7Z41Dfv,FVNhqQKi9PA1jM0kzUoCsQ" -->
+
+## Intro
+
+This section of the docs looks more closely at the full topology of Polygon's zero-knowledge system architecture. 
+
+Built for CDK stacks and using zkEVM technology, these documents explain the contents, layout, and interactions between component systems and functions, drilling down into detailed descriptions of the components that make up the wider system with reference to the relevant code bases.
```
Comment (hsutaiyu) [inline docs/zkEVM/architecture/high-level/overview.md]:
```suggestion
Tailored for CDK stacks and using zkEVM technology, these documents explore the contents, layout, and interactions among component systems and functions. They offer in-depth descriptions of the components that constitute the larger system, with references to relevant code bases.
```

File: docs/zkEVM/architecture/high-level/overview.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+The diagram below is a full and detailed topological overview of the entire Polygon zero-knowledge system architecture.
+
+## Architectural topology
+
+![Polygon systems topology](../../../img/cdk/high-level-architecture/full-topology.png)
+
+## Components 
+
+- Smart contracts: L1 and L2 Solidity smart contracts used for interacting with the whole stack. See the discussion on [zkEVM smart contracts](../../architecture/protocol/zkevm-bridge/smart-contracts.md) for more information.
```
Comment (hsutaiyu) [inline docs/zkEVM/architecture/high-level/overview.md:17]:
```suggestion
- Smart contracts: L1 and L2 Solidity smart contracts used for interacting with the network. See the discussion on [zkEVM smart contracts](../../architecture/protocol/zkevm-bridge/smart-contracts.md) for more information.
```

File: docs/zkEVM/architecture/high-level/overview.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+- Exit root trees: Append-only sparse Merkle trees which record the current and historical state of the system. See the discussion on [zkEVM exit trees](../../architecture/protocol/zkevm-bridge/exit-tree.md) for more information.
+- Aggregator: Used for aggregating transactions into batches for proving. See the discussion on [aggregators](../../architecture/index.md#aggregator) for more information.
```
Comment (hsutaiyu) [inline docs/zkEVM/architecture/high-level/overview.md]:
```suggestion
- Aggregator: Used for aggregating transaction batch sequences for proving. See the discussion on [aggregators](../../architecture/index.md#aggregator) for more information.
```

File: docs/zkEVM/architecture/high-level/overview.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+- Sequencer: Does the complex job of carefully sequencing transactions as they come in before sending them to the aggregator for batching. See the discussion on [sequencers](../../architecture/index.md#sequencer) for more information.
```
Comment (hsutaiyu) [inline docs/zkEVM/architecture/high-level/overview.md]:
```suggestion
- Sequencer: Takes transactions from the pool database and generates batches before sending them to the aggregator. See the discussion on [sequencers](../../architecture/index.md#sequencer) for more information.
```

File: docs/zkEVM/architecture/high-level/overview.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+- Synchronizer: This component ensures a synchronized state between the node's systems and the L1 outside world via the Etherman component and the state database.
```
Comment (hsutaiyu) [inline docs/zkEVM/architecture/high-level/overview.md]: This needs a more detailed description with specifics. Can still be described in one line with the necessary information.

File: docs/zkEVM/architecture/high-level/overview.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+- CDK and zkEVM nodes containing:
+    - Aggregator: Used for aggregating transactions into batches for proving. See the discussion on [aggregators](../../architecture/index.md#aggregator) for more information.
+    - Sequencer: Does the complex job of carefully sequencing transactions as they come in before sending them to the aggregator for batching. See the discussion on [sequencers](../../architecture/index.md#sequencer) for more information.
+    - Synchronizer: This component ensures a synchronized state between the node's systems and the L1 outside world via the Etherman component and the state database. 
+    - Etherman component: The Etherman helps the synchronizer maintain a synchronized state with L1 by communicating with the L1 Ethereum chain via smart contract methods.
+    - JSON RPC client: Allows computational read/write access to the system via published methods which, when called, interact with the pool and state databases.
+    - Pool database: The pool database records transaction requests coming in from the JSON RPC client and sends them to the sequencer.
+    - State database: The state database responds to read requests from the JSON RPC client.
```
Comment (hsutaiyu) [inline docs/zkEVM/architecture/high-level/overview.md:22]: This definition needs to briefly cover what state information is stored in the DB and how it is used by the network.

File: docs/zkEVM/architecture/high-level/overview.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+- Bridge service component: Main facility for transferring tokens from L1 to L2 and back again.
+- Prover component: System for calculating zero-knowledge proofs on transaction batches.
+
+## What to expect
+
+When complete, this section will include information on: 
+
+- The structure of a CDK node and how it interacts with L2 and L1 smart contracts.
+- The structure of a zkEVM node and how it interacts with L2 and L1 smart contracts.
+- The key components included in the nodes, how they function, and their interactions with other components, external developers, and the L1/L2 environment.
```
Comment (hsutaiyu) [inline docs/zkEVM/architecture/high-level/overview.md]:
```suggestion
- The key components included in the nodes, how they function, and their interactions with other components, external dApps, and the L1/L2 environment.
```

File: docs/zkEVM/architecture/high-level/overview.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+- Key similarities and differences between a CDK and zkEVM node set up.
```
Comment (hsutaiyu) [inline docs/zkEVM/architecture/high-level/overview.md]:
```suggestion
- Key similarities and differences between a CDK and zkEVM node.
```

**Review by hsutaiyu — APPROVED** [review]
Approved with fixes and suggestions.

**Comment by hsutaiyu** [comment]
Should we change the directory name? It still says "hermez-node"
Might cause links to break though.

File: docs/zkEVM/architecture/high-level/overview.md [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+## Architectural topology
+
+![Polygon systems topology](../../../img/cdk/high-level-architecture/full-topology.png)
+
+## Components 
+
+- Smart contracts: L1 and L2 Solidity smart contracts used for interacting with the whole stack. See the discussion on [zkEVM smart contracts](../../architecture/protocol/zkevm-bridge/smart-contracts.md) for more information.
+- Exit root trees: Append-only sparse Merkle trees which record the current and historical state of the system. See the discussion on [zkEVM exit trees](../../architecture/protocol/zkevm-bridge/exit-tree.md) for more information.
+- CDK and zkEVM nodes containing:
+    - Aggregator: Used for aggregating transactions into batches for proving. See the discussion on [aggregators](../../architecture/index.md#aggregator) for more information.
```
Comment (kmurphypolygon) [inline docs/zkEVM/architecture/high-level/overview.md]:
```suggestion
    - Aggregator: Used for aggregating transaction batches to send to the prover. See the discussion on [aggregators](../../architecture/index.md#aggregator) for more information.
```

File: docs/zkEVM/architecture/high-level/overview.md (hunk 1 above)
Comment (kmurphypolygon) [reply docs/zkEVM/architecture/high-level/overview.md]: this was outdated, i included some of your suggestions

File: docs/zkEVM/architecture/high-level/overview.md (hunk 6 above)
Comment (kmurphypolygon) [reply docs/zkEVM/architecture/high-level/overview.md]: we are going into detail later, please make a suggestion ...

File: docs/zkEVM/architecture/high-level/overview.md (hunk 7 above)
Comment (kmurphypolygon) [reply docs/zkEVM/architecture/high-level/overview.md:22]: we are going into detail later

**Comment by kmurphypolygon** [comment]
> Should we change the directory name? It still says "hermez-node" Might cause links to break though.

this is in reference to the pic naming ... i can change it now and not break any links as nothing is published yet ... what do you suggest? zkevm-node maybe

File: docs/zkEVM/architecture/high-level/overview.md [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+
+The diagram below is a full and detailed topological overview of the entire Polygon zero-knowledge system architecture.
+
+## Architectural topology
+
+![Polygon systems topology](../../../img/cdk/high-level-architecture/full-topology.png)
+
+## Components 
+
+- Smart contracts: L1 and L2 Solidity smart contracts used for interacting with the whole stack. See the discussion on [zkEVM smart contracts](../../architecture/protocol/zkevm-bridge/smart-contracts.md) for more information.
```
Comment (hsutaiyu) [inline docs/zkEVM/architecture/high-level/overview.md:17]:
```suggestion
- Smart contracts: L1 contracts that manage core functionalities, and L2 contracts that facilitate faster transaction execution and scalability. See the discussion on [zkEVM smart contracts](../../architecture/protocol/zkevm-bridge/smart-contracts.md) for more information.
```

File: docs/zkEVM/architecture/high-level/overview.md [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+- Exit root trees: Append-only sparse Merkle trees which record the current and historical state of the system. See the discussion on [zkEVM exit trees](../../architecture/protocol/zkevm-bridge/exit-tree.md) for more information.
+- CDK and zkEVM nodes containing:
+    - JSON RPC client: Allows computational read/write access to the system via published methods which, when called, interact with the pool and state databases.
+    - Pool database: The pool database records transaction requests coming in from the JSON RPC client and sends them to the sequencer.
```
Comment (hsutaiyu) [inline docs/zkEVM/architecture/high-level/overview.md:21]:
```suggestion
    - Pool database: The pool database, also referred to as the mempool, records transaction requests coming in from the JSON RPC client and sends them to the sequencer.
```

File: docs/zkEVM/architecture/high-level/overview.md [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+    - State database: The state database responds to read requests from the JSON RPC client.
```
Comment (hsutaiyu) [inline docs/zkEVM/architecture/high-level/overview.md:22]:
```suggestion
    - State database: The state database stores state data and responds to read requests from the JSON RPC client.
```

File: docs/zkEVM/architecture/high-level/overview.md [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+    - Sequencer: Does the complex job of carefully sequencing transactions as they come in before sending them to the aggregator for batching. See the discussion on [sequencers](../../architecture/index.md#sequencer) for more information.
```
Comment (hsutaiyu) [inline docs/zkEVM/architecture/high-level/overview.md:23]:
```suggestion
    - Sequencer: Receives transactions from the mempool, orders them, generates transaction batches, and sends them to the aggregator. See the discussion on [sequencers](../../architecture/index.md#sequencer) for more information.
```

File: docs/zkEVM/architecture/high-level/overview.md [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+    - Aggregator: Used for aggregating transaction batches to send to the prover. See the discussion on [aggregators](../../architecture/index.md#aggregator) for more information.
```
Comment (hsutaiyu) [inline docs/zkEVM/architecture/high-level/overview.md:24]:
```suggestion
    - Aggregator: Aggregates batch sequences and sends them to the prover. See the discussion on [aggregators](../../architecture/index.md#aggregator) for more information.
```

File: docs/zkEVM/architecture/high-level/overview.md [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+    - Synchronizer: This component ensures a synchronized state between the node's systems and the L1 outside world via the Etherman component and the state database.
```
Comment (hsutaiyu) [inline docs/zkEVM/architecture/high-level/overview.md]:
```suggestion
    - Synchronizer: Supports co-ordination between contracts and off-chain zkEVM execution environment ensuring integrity of state transitions via the Etherman component and the state database.
```

File: docs/zkEVM/architecture/high-level/overview.md [hunk 17]
```diff
@@ -0,0 +1,45 @@
...
+    - Pool database: The pool database records transaction requests coming in from the JSON RPC client and sends them to the sequencer.
+    - State database: The state database responds to read requests from the JSON RPC client.
+    - Sequencer: Does the complex job of carefully sequencing transactions as they come in before sending them to the aggregator for batching. See the discussion on [sequencers](../../architecture/index.md#sequencer) for more information.
+    - Aggregator: Used for aggregating transaction batches to send to the prover. See the discussion on [aggregators](../../architecture/index.md#aggregator) for more information.
+    - Synchronizer: This component ensures a synchronized state between the node's systems and the L1 outside world via the Etherman component and the state database. 
+    - Etherman component: The Etherman helps the synchronizer maintain a synchronized state with L1 by communicating with the L1 Ethereum chain via smart contract methods.
+- Bridge service component: Main facility for transferring tokens from L1 to L2 and back again.
```
Comment (hsutaiyu) [inline docs/zkEVM/architecture/high-level/overview.md]:
```suggestion
- Bridge service component: Main facility for transferring tokens between L1 and L2.
```

**Review by hsutaiyu — COMMENTED** [review]
Additional suggestions

File: docs/zkEVM/architecture/high-level/overview.md (hunk 11 above)
Comment (kmurphypolygon) [reply docs/zkEVM/architecture/high-level/overview.md:17]: I don't think the Solidity facilitates faster transaction and scalability, that would be the core processing.

**Pull request merged**